thorlcr/activities/join/thjoinslave.cpp

```diff
@@ -444,10 +444,7 @@ class JoinSlaveActivity : public CSlaveActivity, implements ILookAheadStopNotify
             leftInput->getMetaInfo(info);
             if (info.totalRowsMax==0) 
                 isemptylhs = true;
-            if (rightpartition)
-                leftStream.set(leftInputStream); // already ungrouped
-            else
-                leftStream.setown(createUngroupStream(leftInputStream));
+            leftStream.setown(createUngroupStream(leftInputStream));
```

Comment:
Is there any way of only doing this if the stream is grouped?

Reply:
I guess it is an extra virtual call per row,
and I think I can test input->isGrouped() to make it conditional